src/Microsoft.AspNet.WebHooks.Custom.SqlStorage/WebHooks/WebHookStoreContext.cs

```diff
@@ -12,16 +15,62 @@ namespace Microsoft.AspNet.WebHooks
     /// </summary>
     public class WebHookStoreContext : DbContext
     {
-        internal const string ConnectionStringName = "MS_SqlStoreConnectionString";
-        private const string ConnectionStringNameParameter = "name=" + ConnectionStringName;
+        private readonly string _tableName;
+        private readonly string _schemaName = "WebHooks";
+        internal static string ConnectionStringName = "MS_SqlStoreConnectionString";
 
         /// <summary>
         /// Initializes a new instance of the <see cref="WebHookStoreContext"/> class.
         /// </summary>
-        public WebHookStoreContext() : base(ConnectionStringNameParameter)
+        public WebHookStoreContext()
+            : base(GetConnectionStringNameParameter(ConnectionStringName))
         {
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebHookStoreContext"/> class using the given string
+        /// as the name or connection string for the database to which a connection will be made.
+        /// </summary>
+        /// <param name="nameOrConnectionString">Either the database name or a connection string.</param>
+        public WebHookStoreContext(string nameOrConnectionString)
+            : base(GetConnectionStringNameParameter(nameOrConnectionString))
```

Comment:
This ties the `HttpConfigurationExtensions` caller's hands and needs to go away. We specifically want users to be able to pass the connection string itself into `InitializeCustomWebHooksSqlStorage(...)`.

Remove `GetConnectionStringNameParameter(...)`, restore `ConnectionStringNameParameter`, and pass whatever `nameOrConnectionString` value this is given directly to the base `DbContext`.

Reply:
Then it's not make sense to check `nameOrConnectionString` because it will be done by `DbContext`